agent/structs/intention.go

```diff
@@ -641,6 +641,17 @@ type IntentionQueryCheckResponse struct {
 	Allowed bool
 }
 
+// IntentionSummary contains a summary of a set of intentions between two services
+// Currently contains:
+// - Whether all actions are allowed
+// - Whether the matching intention has L7 permissions attached
+// - Whether the intention is managed by an external source like k8s,
+type IntentionDecisionSummary struct {
+	Allowed          bool
+	HasL7Permissions bool
```

Comment:
I thought I heard we decided on `HasPermissions` for this somewhere? is the L7 part important? I don't really mind TBH it's only an internal endpoint but would be good to choose so we can build the mock right etc.